pkg/sidecar/mock.go

```diff
@@ -0,0 +1,118 @@
+package sidecar
```

Comment:
I have this in the "sidecar" package rather than the "sidecar_test" package and I am a bit torn about this.

I need this to be in the sidecar package in order to use the unexported handler, and as a side benefit we may be able to use this in CI with the real nonInMem sync service if we want to. It would be a NoOp -- but still sync-service-using mode. 

On the other hand, maybe using this in CI isn't all that useful, and by moving it to sidecar_test we could avoid having this compiled in.

@nonsense what do you think?